libs/components/forms/src/lib/modules/file-attachment/file-attachment.component.spec.ts

```diff
@@ -1351,11 +1376,64 @@ describe('File attachment', () => {
   });
 
   it('should pass accessibility when label does not match the button text', async () => {
-    fixture.componentInstance.labelText = 'Something different';
+    fixture.componentInstance.labelElementText = 'Something different';
     fixture.detectChanges();
     await fixture.whenStable();
     await expectAsync(fixture.nativeElement).toBeAccessible();
   });
+
+  it('should pass accessibility when `labelText` is set', async () => {
+    fixture.componentInstance.labelText = 'Attach file';
+    fixture.componentInstance.labelElementText = undefined;
+    fixture.detectChanges();
+
+    await expectAsync(fixture.nativeElement).toBeAccessible();
+  });
+
+  it('should render `labelText` and not label element if `labelText` is set', async () => {
+    fixture.componentInstance.labelElementText = 'label element';
+    fixture.componentInstance.labelText = 'label text';
+    fixture.detectChanges();
+
+    const label = getLabelWrapper();
```

Comment:
This line and the line below it are repeated multiple times in subsequent tests, and the logic in them is not trivial. Factor it out into a function:

```
function validateLabelText(expectedText: string): void {
    const label = getLabelWrapper();
    expect(label?.textContent?.trim()).toBe(expectedText);
}
```